src/TLuaInterpreter.cpp

```diff
@@ -3657,61 +3657,37 @@ int TLuaInterpreter::getMudletInfo(lua_State* L)
 {
     Host& host = getHostFromLua(L);
 
-    QByteArrayList knownEncodings{"\"ASCII\""};
-    auto adjustEncoding = [](auto encodingName) {
-        auto originalEncoding = encodingName;
-        if (encodingName.startsWith("M_")) {
-            encodingName.remove(0, 2);
+    QStringList knownEncodings{"ASCII"};
+    QString currentEncoding{host.mTelnet.getEncoding()};
+    {
+        auto adjustEncoding = [](auto encodingName) {
+            auto originalEncoding = encodingName;
+            if (encodingName.startsWith("M_")) {
+                encodingName.remove(0, 2);
+            }
+
+            return (originalEncoding == encodingName) ? originalEncoding : QStringLiteral("%1 (%2)").arg(encodingName, originalEncoding);
+        };
+        for (const auto& encoding : host.mTelnet.getEncodingsList()) {
+            knownEncodings.append(QString(adjustEncoding(encoding)));
         }
+        knownEncodings.sort(Qt::CaseInsensitive);
```

Comment:
That isn't a good idea - and the list is already sorted into a sensible order (ASCII, UTF-8 and other multi-byte encodings - and using a simple sort on the `ISO 8859-X` ones doesn't do the sort that would be wanted to put `-9` before `-10`. Specifying a case consideration is also pointless as they are all upper case as well.

Reply:
That sort of sort is not going to put these in ascending number order:
```
ISO 8859-1
ISO 8859-10
ISO 8859-11
ISO 8859-13
ISO 8859-14
ISO 8859-15
ISO 8859-16
ISO 8859-2
ISO 8859-3
ISO 8859-4
ISO 8859-5
ISO 8859-6
ISO 8859-7
ISO 8859-8
ISO 8859-9
```

Reply:
Fair - fixed